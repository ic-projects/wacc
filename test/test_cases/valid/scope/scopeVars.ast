-- Test: scopeVars.wacc

-- Uploaded file: 
---------------------------------------------------------------
# simple variable scoping test

# Output:
# 2
# 4
# 2

# Program:

begin
  int x = 2 ;
  println x ;
  begin
    int x = 4 ;
    println x
  end ;
  println x 
end

---------------------------------------------------------------

-- Compiler Output:
-- Compiling...
-- Printing AST...
scopeVars.ast contents are:
===========================================================
0	Program
1	  - int main()
2	    - DECLARE
3	      - TYPE
4	        - int
5	      - LHS
6	        - x
7	      - RHS
8	        - 2
9	    - PRINTLN
10	      - x
11	    - SCOPE
12	      - DECLARE
13	        - TYPE
14	          - int
15	        - LHS
16	          - x
17	        - RHS
18	          - 4
19	      - PRINTLN
20	        - x
21	    - PRINTLN
22	      - x
===========================================================
-- Finished

