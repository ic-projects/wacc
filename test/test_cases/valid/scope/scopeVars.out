-- Compiling...
-- Assembling and Linking...
-- Executing...
===========================================================
2
4
2
===========================================================
The exit code is 0.
-- Finished
